lib/jekyll/commands/compose.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+module Jekyll
+  module Commands
+    class ComposeCommand < Command
+      def self.init_with_program(prog)
+        prog.command(:compose) do |c|
+          c.syntax "compose [--post|--draft|--collection COLLECTION] NAME"
+          c.description "Creates a new document with the given NAME"
+
+          options.each { |opt| c.option *opt }
+
+          c.action { |args, options| process args, options }
+        end
+      end
+
+      def self.options
+        [
+          ["extension", "-x EXTENSION", "--extension EXTENSION", "Specify the file extension"],
+          ["layout", "-l LAYOUT", "--layout LAYOUT", "Specify the document layout"],
+          ["force", "-f", "--force", "Overwrite a document if it already exists"],
+          ["date", "-d DATE", "--date DATE", "Specify the document date"],
+          ["collection", "-c COLLECTION", "--collection COLLECTION", "Specify the document collection"],
+          ["post", "--post", "Create a new post (default)"],
+          ["draft", "--draft", "Create a new draft"],
+          ["config", "--config CONFIG_FILE[,CONFIG_FILE2,...]", Array, "Custom configuration file"],
+        ]
+      end
+
+      def self.process(args = [], options = {})
+        config = configuration_from_options(options)
+        params = ComposeCommandArgParser.new args, options, config
+        params.validate!
+
+        document = ComposeCommandFileInfo.new params
+
+        file_creator = Compose::FileCreator.new(document, params.force?, params.source)
+        file_creator.create!
+
+        Compose::FileEditor.bootstrap(config)
+        Compose::FileEditor.open_editor(file_creator.file_path)
+      end
+
+      class ComposeCommandArgParser < Compose::ArgParser
+        def validate!
+          if options.values_at("post", "draft", "collection").compact.length > 1
+            raise ArgumentError, "You can only specify one of --post, --draft, or --collection COLLECTION."
+          end
+
+          super
+        end
+
+        def date
+          options["date"] ? Date.parse(options["date"]) : Time.now
+        end
+
+        def collection
+          options["collection"] || (options["post"] ? "posts" : nil) || (options["draft"] ? "drafts" : nil) || "posts"
```

Comment:
Whoa! This is not equivalent to the previous implementation.
If this is the route you intend to follow, please use conventional `if .. elsif` syntax.

Reply:
This is the change you requested https://github.com/jekyll/jekyll-compose/pull/103#discussion_r321104930

Reply:
This was the previous implementation, which I see doesn't line up with "negative nil-checks". I think the comment may not have been attached to the right code there. 

What change did you want, if any, for this?

```
      class ComposeCommandFileInfo < Compose::FileInfo
        def resource_type
          case @params.collection
          when "posts"
            "post"
          when "drafts"
            "draft"
          else
            "file"
          end
        end
```

Reply:
I *suggested* a simpler version:
```ruby
options["collection"] || options["post"] || options["draft"] || "posts"
```

If that wont work, then prefer using conventional syntax for readability.

Reply:
`options["post"]` is a flag (no argument) and so is a value of `true`, not `"posts"`, thus the need for the extra code. I'll revert the change.